fallback/s2a_fallback_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package fallback
+
+import (
+	"testing"
+)
+
+func TestDefaultFallbackClientHandshakeFunc(t *testing.T) {
+	for _, tc := range []struct {
+		desc              string
+		inputFallbackAddr string
+		funcInitialized bool
+		expectNilErr	bool
+	}{
+		{
+			"working case, fallback address has port suffix",
+			"example.com:443",
+			true,
+			true,
+		},
+		{
+			"working case, fallback address has no port suffix",
+			"example.com",
+			true,
+			true,
+		},
+		{
+			"test empty fallback address",
+			"",
+			false,
+			false,
+		},
+	} {
+		fbFunc, err := DefaultFallbackClientHandshakeFunc(tc.inputFallbackAddr)
```

Comment:
Both here and below, we should be executing the fallback function or fallback dialer in the tests. :)